packages/region/src/services/region-module.ts

```diff
@@ -188,26 +190,36 @@ export default class RegionModuleService<
   }
 
   async update(
-    data: UpdateRegionDTO[],
+    selector: FilterableRegionProps,
+    data: UpdatableRegionFields,
     sharedContext?: Context
   ): Promise<RegionDTO[]>
   async update(
-    data: UpdateRegionDTO,
+    regionId: string,
+    data: UpdatableRegionFields,
     sharedContext?: Context
   ): Promise<RegionDTO>
   @InjectTransactionManager("baseRepository_")
   async update(
-    data: UpdateRegionDTO | UpdateRegionDTO[],
+    idOrSelector: string | FilterableRegionProps,
```

Comment:
suggestion: being able to provide an array of dto would be helpful for most cases like you your steps

Reply:
Think we should use upsert for this instead

Reply:
but if you don't want to create but only update, if it does not exists you might not want to create it under the hood but maybe be aware that it does not exists no?

Reply:
for example, in the workflow during the rollback we update them, if for some reason, they have been deleted elsewhere, you might re create them accidentally when you didn't want to. So the update in that case would be better than an upsert. Also, it is common in our modules that the update takes an array of data that includes the primary key(s) to perform the bulk update

Reply:
Perhaps, we should formalize the signatures, so we know exactly when to use what method and how to implement them.

@adrien2p, can you let me know if the approach taken in the latest commit is along the lines of what you would like to see?

Reply:
Yes it looks like it is 👍